connectors/migrations/20251120_migrate_slack_connection_credentials.ts

```diff
@@ -0,0 +1,237 @@
+  api: OAuthAPI,
+  connector: ConnectorResource,
+  slackClientId: string,
+  slackClientSecret: string,
+  slackSigningSecret: string,
+  logger: Logger,
+  execute: boolean
+): Promise<Result<void, OAuthAPIError>> {
+  logger.info(
+    `Migrating connection credential for connector ${connector.id}, connectionId ${connector.connectionId}.`
+  );
+
+  if (!execute) {
+    logger.info(
+      `[DRY RUN] Would create credential and update connection ${connector.connectionId}.`
+    );
+    return new Ok(undefined);
+  }
+
+  // Get connection metadata to extract user_id
+  const metadataRes = await api.getConnectionMetadata({
+    connectionId: connector.connectionId,
+  });
+
+  if (metadataRes.isErr()) {
+    logger.error(
+      { error: metadataRes.error },
+      "Failed to fetch connection metadata."
+    );
+    return metadataRes;
+  }
+
+  const userId = metadataRes.value.connection.metadata.user_id;
+  const teamId = metadataRes.value.connection.metadata.team_id;
+
+  if (typeof userId !== "string" || userId.length === 0) {
+    return new Err({
+      code: "missing_user_id",
+      message: `Connection ${connector.connectionId} has no user_id in metadata.`,
+    });
+  }
+
+  if (typeof teamId !== "string" || teamId.length === 0) {
+    return new Err({
+      code: "missing_team_id",
+      message: `Connection ${connector.connectionId} has no team_id in metadata.`,
+    });
+  }
+
+  logger.info(
+    { userId, teamId },
+    "Using user_id and team_id from connection metadata."
+  );
+
+  // Create a credential record for this connection
+  const credRes = await api.postCredentials({
+    provider: PROVIDER,
+    userId,
+    workspaceId: connector.workspaceId,
+    credentials: {
+      client_id: slackClientId,
+      client_secret: slackClientSecret,
+    },
+  });
+
+  if (credRes.isErr()) {
+    return credRes;
+  }
+
+  const credentialId = credRes.value.credential.credential_id;
+  logger.info(
+    { credentialId },
+    `Created credential for connector ${connector.id}.`
+  );
+
+  // Update the connection to reference this credential and set client_id in metadata
+  const updateRes = await api.updateConnectionCredential({
+    connectionId: connector.connectionId,
+    relatedCredentialId: credentialId,
+    metadata: { client_id: slackClientId },
+  });
+
+  if (updateRes.isErr()) {
+    return updateRes;
+  }
+
+  logger.info(
+    { credentialId },
+    `Successfully updated connection ${connector.connectionId} to use credential and set client_id in metadata.`
+  );
+
+  // Store credential reference in SlackConfiguration
+  const slackConfig = await SlackConfigurationResource.fetchByConnectorId(
+    connector.id
+  );
+
+  if (!slackConfig) {
+    return new Err({
+      code: "missing_slack_config",
+      message: `No Slack configuration found for connector ${connector.id}`,
+    });
+  }
+
+  await SlackConfigurationModel.update(
+    { privateIntegrationCredentialId: credentialId },
```

Comment:
nit: `privateIntegrationCredentialId` name becomes confusing, as it's now for all integrations.